trace/trace.go

```diff
@@ -42,6 +44,18 @@ const (
 
 	errInvalidSpanIDLength errorConst = "hex encoded span-id must have length equals to 16"
 	errNilSpanID           errorConst = "span-id can't be all zero"
+
+	// based on the W3C Trace Context specification, see https://www.w3.org/TR/trace-context-1/#tracestate-header
+	traceStateKeyFormat                      = `[a-z][_0-9a-z\-\*\/]{0,255}`
+	traceStateKeyFormatWithMultiTenantVendor = `[a-z][_0-9a-z\-\*\/]{0,240}@[a-z][_0-9a-z\-\*\/]{0,13}`
```

Comment:
If I understand [the specification](https://www.w3.org/TR/trace-context-1/#key) this can start with a digit as well:

> key = ( lcalpha / DIGIT ) 0*240( lcalpha / DIGIT / "_" / "-"/ "*" / "/" ) "@" lcalpha 0*13( lcalpha / DIGIT / "_" / "-"/ "*" / "/" )

Reply:
Good catch :+1:

Reply:
Looks like this is also not 100% clear, looking at your issue at https://github.com/w3c/trace-context/issues/441. Let's see after clarification.

Reply:
True. I'm fine with leaving this as is. Given I opened the w3C issue I'll be sure to know what the final decision is and can open an follow on issue if needed.